### PR TITLE
Adding a why property to a queueitem object

### PR DESCRIPTION
**Pull request by tmcarr**
I'm currently using this library and noticed that it didn't have a way to read the "why" field of the QueueItem object without referring to _data directly.

This patch adds a property to make this cleaner. 

I didn't see any tests for this, so I haven't added any here. It's a pretty straightforward change.

### CODE REVIEW

**Comment by tmcarr** [comment]
Looks like some of the tests failed, but the failures seem unrelated to my change.

**Comment by lechat** [comment]
Thanks for this!

Does that field exist in both, 1.x and 2.x Jenkins? 

Would you mind adding a test for that field access? 

The failing tests are due to incompatibility of credentials REST API between 1.x and 2.x, I will fix that once I will have time.

**Comment by tmcarr** [comment]
I know for a fact that Jenkins 1.5 and 1.6 return this. (I'm adding this feature so we can use it with Twitter's 1.6 install.) 

As far as newer Jenkins goes, I fired up Jenkins 2.3 in a docker container and ran this against it:

`from jenkinsapi.jenkins import Jenkins
from pprint import pprint

master = Jenkins('http://localhost:8080')
queue = master.get_queue()

print("Number of jobs in queue: %s" % len(queue))
for build, data in queue.iteritems():
  pprint(data._data)
`
And the output still has a why field:

`Number of jobs in queue: 1
{'_class': 'hudson.model.Queue$BuildableItem',
 'actions': [{'_class': 'hudson.model.CauseAction',
              'causes': [{'_class': 'hudson.model.Cause$UserIdCause',
                          'shortDescription': 'Started by user Travis Carr',
                          'userId': 'tmcarr',
                          'userName': 'Travis Carr'}]}],
 'blocked': False,
 'buildable': True,
 'buildableStartMilliseconds': 1466622044578,
 'id': 1,
 'inQueueSince': 1466622044571,
 'params': '',
 'pending': False,
 'stuck': True,
 'task': {'_class': 'hudson.model.FreeStyleProject',
          'color': 'notbuilt',
          'name': 'testing',
          'url': 'http://localhost:8080/job/testing/'},
 'url': 'queue/item/1/',
 'why': 'Waiting for next available executor'}`

Working on the tests now.

**Comment by tmcarr** [comment]
I'm having trouble actually testing my tests because they fail locally... Do you know of a sane way to test them in isolation, or even better, against my own copy of jenkins? I have a working jenkins docker instance on my laptop I can point it at...

**Comment by tmcarr** [comment]
Oh hey... Figured out the issue with local tests. Now I can stop spamming your TravisCI instance.

File: jenkinsapi_tests/systests/test_queue.py [hunk 1]
```diff
@@ -79,6 +79,22 @@ def test_start_and_stop_long_running_job(self):
...
+        job_names = [random_string() for _ in range(2)]
+        jobs = []
+        for job_name in job_names:
+            j = self.jenkins.create_job(job_name, LONG_RUNNING_JOB)
+            jobs.append(j)
+            j.invoke()
+
+        queue = self.jenkins.get_queue()
+        for _, item in queue.iteritems():
+            assert item.why() is StringType
```
Comment (lechat) [inline jenkinsapi_tests/systests/test_queue.py]: ``why`` is marked as ``property``, why it is used as a function here?

Comment (tmcarr) [inline jenkinsapi_tests/systests/test_queue.py]: Because I had a brainfart. Incoming patch to get this shippable.

**Comment by tmcarr** [comment]
Cool, looks like the "latest" tests are still failing with the same issue you mentioned before, but everything else passes.

**Pull request merged**